src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/SaveAzureDeploymentTemplateCmdlet.cs

```diff
@@ -0,0 +1,92 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities;
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+    /// <summary>
+    /// Saves the deployment template to a file on disk.
+    /// </summary>
+    [Cmdlet(VerbsData.Save, "AzureRmDeploymentTemplate", SupportsShouldProcess = true,
+        DefaultParameterSetName = SaveAzureDeploymentTemplateCmdlet.DeploymentNameParameterSet), OutputType(typeof(PSObject))]
```

Comment:
@Tiano2017 same comment about using a concrete type instead of `PSObject`

Reply:
This one should be fine. The output is just about a file path and we don't expect this to be in a pipeline.

Reply:
@Tiano2017 even if we are returning the file path, we should still have a concrete type with that path as one of its properties -- we are now generating docs for our types, which lets the user know what properties and methods are exposed on parameter and output types, and it is also best practice to use concrete .NET types in PowerShell for consistency in returned objects

Reply:
Fixed. 

---
In reply to: [205822769](https://github.com/Azure/azure-powershell/pull/6516#discussion_r205822769) [](ancestors = 205822769)